Mathlib/Tactic/NormNum/Basic.lean

```diff
@@ -230,18 +234,34 @@ such that `norm_num` successfully recognises both `a` and `b`. -/
 @[norm_num _ - _, Sub.sub _ _] def evalSub : NormNumExt where eval {u α} e := do
   let .app (.app f (a : Q($α))) (b : Q($α)) ← whnfR e | failure
   let ra ← derive a; let rb ← derive b
-  let rα ← inferRing α
-  guard <|← withNewMCtxDepth <| isDefEq f q(HSub.hSub (α := $α))
+  let rα? ← inferRing? α
+  if let some _ := rα? then
+    guard <|← withNewMCtxDepth <| isDefEq f q(HSub.hSub (α := $α))
+  else
+    -- We trust that the default instance for `HSub` is `Nat.sub` when the first parameter is `ℕ`.
+    guard <|← withNewMCtxDepth <| isDefEq f q(HSub.hSub (α := ℕ))
```

Comment:
The Nat.sub evaluator should be a separate norm num extension from this one.

Reply:
Ok, separated! (git history rewritten, since there are only a couple commits)

New open question about a maybe-unnecessary check marked with `!!`.